Description: Fixating FreeType2 include directory
Author: Francesco Paolo Lovergine <frankie@debian.org>
Last-Update: 2013-12-11
--- mapserver.orig/cmake/FindFreetype.cmake	2013-12-11 16:33:03.000000000 +0100
+++ mapserver/cmake/FindFreetype.cmake	2013-12-11 16:35:49.000000000 +0100
@@ -52,9 +52,10 @@
   /sw/include
   /opt/local/include
   /usr/freeware/include
+  /usr/include/freetype2
 )
 
-FIND_PATH(FREETYPE_INCLUDE_DIR_freetype2 freetype/config/ftheader.h 
+FIND_PATH(FREETYPE_INCLUDE_DIR_freetype2 config/ftheader.h 
   HINTS
   $ENV{FREETYPE_DIR}/include/freetype2
   PATHS
@@ -65,6 +66,7 @@
   /opt/local/include
   /usr/freeware/include
   PATH_SUFFIXES freetype2
+  /usr/include/freetype2
 )
 
 set(FREETYPE_NAMES ${FREETYPE_NAMES} freetype libfreetype freetype219 freetype239 freetype241MT_D freetype2411)
