From 6aa5873862fb86b2b418de990e0b1319ff05b9e7 Mon Sep 17 00:00:00 2001
From: Thomas Bonfort <thomas.bonfort@gmail.com>
Date: Thu, 9 Jul 2015 15:24:32 +0200
Subject: [PATCH] missing mapows.h includes

---
 mapcontext.c       | 2 +-
 mapdraw.c          | 1 +
 mapkmlrenderer.cpp | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

--- a/mapcontext.c
+++ b/mapcontext.c
@@ -27,7 +27,7 @@
  ****************************************************************************/
 
 #include "mapserver.h"
-
+#include "mapows.h"
 
 
 #if defined(USE_WMS_LYR) && defined(USE_OGR)
--- a/mapdraw.c
+++ b/mapdraw.c
@@ -33,6 +33,7 @@
 #include "maptime.h"
 #include "mapcopy.h"
 #include "mapfile.h"
+#include "mapows.h"
 
 
 /* msPrepareImage()
--- a/mapkmlrenderer.cpp
+++ b/mapkmlrenderer.cpp
@@ -34,6 +34,7 @@
 #include "maperror.h"
 #include "mapkmlrenderer.h"
 #include "mapio.h"
+#include "mapows.h"
 
 #if defined(USE_OGR)
 #  include "cpl_conv.h"
