Description: Fix variables set by find_package(PHP).
Author: Bas Couwenberg <sebastic@debian.org>
Forwarded: https://github.com/mapserver/mapserver/pull/5776

--- a/mapscript/phpng/CMakeLists.txt
+++ b/mapscript/phpng/CMakeLists.txt
@@ -22,17 +22,17 @@ add_definitions(-DZTS=1)
 endif(WITH_THREAD_SAFETY)
 endif(WIN32)
 
-include_directories(${PHP_FOUND_INCLUDE_PATH})
-include_directories(${PHP_FOUND_INCLUDE_PATH}/main)
-include_directories(${PHP_FOUND_INCLUDE_PATH}/Zend)
-include_directories(${PHP_FOUND_INCLUDE_PATH}/TSRM)
+include_directories(${PHP_INCLUDE_PATH})
+include_directories(${PHP_INCLUDE_PATH}/main)
+include_directories(${PHP_INCLUDE_PATH}/Zend)
+include_directories(${PHP_INCLUDE_PATH}/TSRM)
 
 include_directories(${PROJECT_SOURCE_DIR}/mapscript/swiginc)
 include_directories(${PROJECT_SOURCE_DIR}/mapscript/)
 include_directories(${PROJECT_SOURCE_DIR}/mapscript/phpng)
 
 if(WIN32)
-include_directories(${PHP_FOUND_INCLUDE_PATH}/win32)
+include_directories(${PHP_INCLUDE_PATH}/win32)
 endif(WIN32)
 
 IF(PHP_VERSION LESS 5)
@@ -56,4 +56,4 @@ target_link_libraries(php_mapscriptng ma
 set_target_properties(${SWIG_MODULE_mapscript_REAL_NAME} PROPERTIES PREFIX "")
 if(NOT WIN32)
 install(TARGETS php_mapscriptng DESTINATION ${PHP_EXTENSION_DIR})
-endif()
\ No newline at end of file
+endif()
