Description: install required mapscript.php
Author: Jeff McKenna <jmckenna@gatewaygeomatics.com>
Origin: https://github.com/mapserver/mapserver/pull/6104
Bug: https://bugs.launchpad.net/ubuntu/+source/mapserver/+bug/1884854

--- a/mapscript/phpng/CMakeLists.txt
+++ b/mapscript/phpng/CMakeLists.txt
@@ -84,4 +84,6 @@ set_target_properties(${SWIG_MODULE_maps
 
 if(NOT WIN32)
   install(TARGETS php_mapscriptng DESTINATION ${PHP_EXTENSION_DIR})
+  # install the required file containing MapServer constants and functions
+  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/mapscript.php DESTINATION ${PHP_EXTENSION_DIR} COMPONENT dev)
 endif()
