Description: Install Ruby mapscript to vendorarchdir instead of archdir.
Author: Bas Couwenberg <sebastic@debian.org>
Last-Update: 2013-09-11
Forwarded: not-needed

--- a/mapscript/ruby/CMakeLists.txt
+++ b/mapscript/ruby/CMakeLists.txt
@@ -39,5 +39,5 @@ if(APPLE)
 endif(APPLE)
 
 set(mapscript_files $<TARGET_FILE:${SWIG_MODULE_rubymapscript_REAL_NAME}>)
-execute_process(COMMAND ${RUBY_EXECUTABLE} -r rbconfig -e "puts RbConfig::CONFIG['sitearchdir']" OUTPUT_VARIABLE RUBY_SITEARCHDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
-install(FILES ${mapscript_files} DESTINATION ${RUBY_SITEARCHDIR})
+execute_process(COMMAND ${RUBY_EXECUTABLE} -r rbconfig -e "puts RbConfig::CONFIG['vendorarchdir']" OUTPUT_VARIABLE RUBY_VENDORARCHDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
+install(FILES ${mapscript_files} DESTINATION ${RUBY_VENDORARCHDIR})
